### PR TITLE
[queries] Extend several queries

### PR DESCRIPTION
**Pull request by JingsongLi**
At present, nexmark's queries focus too much on window and time join, we need to add some more common patterns.

First, we introduce new fields to bid table: 
- `channel`, it has about 10000 values. The business meaning is the channel from which the bid comes. Then, we can do some aggregation on channels.
- `url`, the url of this bid.

Add 7 cases:
- Query 16: Channel Statistics Report. How many distinct users join the bidding for different level of price for a channel? Illustrates multiple distinct aggregations with filters for multiple keys.
- Query 17: Auction Statistics Report. How many bids on an auction made a day and what is the price? Illustrates an unbounded group aggregation.
- Query 18: Find last bid. What's a's last bid for bidder to auction? Illustrates a Deduplicate query.
- Query 19: Auction TOP-10 Price. What's the top price 10 bids of an auction? Illustrates a TOP-N query.
- Query 20: Expand bid with auction. Get bids with the corresponding auction information where category is 10. Illustrates a filter join.
- Query 21: Add channel id. Add a channel_id column to the bid table. Illustrates a 'CASE WHEN' + 'REGEXP_EXTRACT' SQL.
- Query 22: Get URL Directories. What is the directory structure of the URL? Illustrates a SPLIT_INDEX SQL.

### CODE REVIEW

File: nexmark-flink/src/main/resources/queries/q16.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        count(*) filter (where price >= 10000 and price < 1000000) AS rank2_bids,
+        count(*) filter (where price >= 1000000) AS rank3_bids,
+        count(distinct bidder) AS total_bidders,
+    count(distinct bidder) filter (where price < 10000) AS rank1_bidders,
+        count(distinct bidder) filter (where price >= 10000 and price < 1000000) AS rank2_bidders,
+        count(distinct bidder) filter (where price >= 1000000) AS rank3_bidders,
+        count(distinct auction) AS total_auctions,
+    count(distinct auction) filter (where price < 10000) AS rank1_auctions,
+        count(distinct auction) filter (where price >= 10000 and price < 1000000) AS rank2_auctions,
+        count(distinct auction) filter (where price >= 1000000) AS rank3_auctions
```
Comment (wuchong) [inline nexmark-flink/src/main/resources/queries/q16.sql:45]: format indent?

File: nexmark-flink/src/main/resources/queries/q18.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    url  VARCHAR,
+    dateTime  TIMESTAMP(3),
+    extra  VARCHAR,
+) WITH (
+  'connector' = 'blackhole'
+);
+
+INSERT INTO discard_sink
+SELECT auction, bidder, price, channel, url, dateTime, extra
+ FROM (SELECT *, ROW_NUMBER() OVER (PARTITION BY bidder, auction ORDER BY dateTime) AS rank_number
```
Comment (wuchong) [inline nexmark-flink/src/main/resources/queries/q18.sql:22]: ORDER BY dateTime DESC?

File: nexmark-flink/src/main/resources/queries/q19.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    dateTime  TIMESTAMP(3),
+    extra  VARCHAR,
+    rank  BIGINT
+) WITH (
+  'connector' = 'blackhole'
+);
+
+INSERT INTO discard_sink
+SELECT * FROM
+(SELECT *, ROW_NUMBER() OVER (PARTITION BY auction ORDER BY price) AS rank_number FROM bid)
```
Comment (wuchong) [inline nexmark-flink/src/main/resources/queries/q19.sql:23]: ORDER BY price DESC?

**Review by wuchong — COMMENTED** [review]
Please also update README.

**Comment by JingsongLi** [comment]
Test passed in my local.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**